patroni/dcs/consul.py

```diff
@@ -458,6 +459,7 @@ def _update_service(self, data):
         conn_parts = urlparse(data['conn_url'])
         check = base.Check.http(api_parts.geturl(), self._service_check_interval,
                                 deregister='{0}s'.format(self._client.http.ttl * 10))
+        check['TLSServerName'] = self._service_check_tls_server_name
```

Comment:
We probably don't want to send it if the value is empty and/or missing from the config.

Reply:
> We probably don't want to send it if the value is empty and/or missing from the config.

Consul always sets the value itself anyway, it's an empty string by default (see "example" below), so it doesn't really make a difference (we also use an empty string by default).

If you insist, I could of course add an extra check `if self._service_check_tls_server_name:`, the practical result will be the same.

Example of an existing consul check (configured without specifying TLSServerName):
```json
{
  "Check": {
    "Node": "patroni-node1",
    "CheckID": "service:postgresql/patroni-node1",
    "Name": "Service 'postgresql' check",
    "Status": "critical",
    "Notes": "",
    "Output": "",
    "ServiceID": "postgresql/patroni-node1",
    "ServiceName": "postgresql",
    "ServiceTags": [
      "replica"
    ],
    "Type": "http",
    "Interval": "",
    "Timeout": "",
    "ExposedPort": 0,
    "Definition": {},
    "CreateIndex": 0,
    "ModifyIndex": 0
  },
  "ChkType": {
    "CheckID": "",
    "Name": "",
    "Status": "",
    "Notes": "",
    "ScriptArgs": null,
    "HTTP": "https://10.1.1.1:8008/replica",
    "H2PING": "",
    "Header": null,
    "Method": "",
    "Body": "",
    "TCP": "",
    "Interval": 15000000000,
    "AliasNode": "",
    "AliasService": "",
    "DockerContainerID": "",
    "Shell": "",
    "GRPC": "",
    "GRPCUseTLS": false,
    "TLSServerName": "",
    "TLSSkipVerify": false,
    "Timeout": 0,
    "TTL": 0,
    "SuccessBeforePassing": 0,
    "FailuresBeforeCritical": 0,
    "ProxyHTTP": "",
    "ProxyGRPC": "",
    "DeregisterCriticalServiceAfter": 150000000000,
    "OutputMaxSize": 0
  },
  "Token": "xxx",
  "Source": "remote"
}
```

Reply:
> Consul always sets the value itself anyway


What about compatibility with older Consul versions? For example, v1.9.x doesn't know about `TLSServerName` and I can expect that such requests will fail.

Reply:
@CyberDem0n That's a very good point (wasn't aware that 1.9.x is still a supported release).

Reply:
Changed it, so that it's only set if configured.